Fix hostname argument handling to support IPv6 address family

[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description
[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT share sensitive information, whether personal, proprietary, or otherwise! )
When an IPv6 address is input to the NRI, it does not format as expected by the connector handler (sql.Open). Instead, it needs to be formatted as defined in RFC 2732.

```
[FATAL] can't continue: dial tcp [fc21::200:3306]:3306: getsockopt: no route to host
panic: dial tcp [fc21::200:3306]:3306: getsockopt: no route to host

goroutine 1 [running]:
github.com/newrelic/nri-mysql/vendor/github.com/newrelic/infra-integrations-sdk/log.Fatal(0x781d80, 0xc4200804b0)
        /data/jenkins/infra-integrations-nri-mysql-release-linux-package-2/go/src/github.com/newrelic/nri-mysql/vendor/github.com/newrelic/infra-integrations-sdk/log/log.go:107 +0xd5
main.fatalIfErr(0x781d80, 0xc4200804b0)
        /data/jenkins/infra-integrations-nri-mysql-release-linux-package-2/go/src/github.com/newrelic/nri-mysql/src/mysql.go:133 +0x44
main.main()
        /data/jenkins/infra-integrations-nri-mysql-release-linux-package-2/go/src/github.com/newrelic/nri-mysql/src/mysql.go:96 +0x26d
```

## Expected Behavior

Connect to a remote entity (host) using IPv6 to retrieve metrics.

## Steps to Reproduce
[NOTE]: # ( Please be as specific as possible. )
[TIP]:  # ( Link a sample application that demonstrates the issue. )

Execute command to a MySQL over IPv6 as shown below.

```
$ nri-mysql -hostname fc21::200
```

--- a/src/mysql.go
+++ b/src/mysql.go
@@ -8,6 +8,7 @@ import (
 	"runtime"
 	"strconv"
 	"strings"
+	"net"
 
 	"github.com/newrelic/infra-integrations-sdk/data/attribute"
 
@@ -42,6 +43,17 @@ type argumentList struct {
 	ShowVersion            bool   `default:"false" help:"Print build information and exit"`
 }
 
+func hostnameIpv6Formater(hostname string) (string) {
+	// Format backend parameters
+	ip := net.ParseIP(hostname)
+
+	if ip.To16() != nil {
+		return fmt.Sprintf("[%v]", ip)
+	}
+
+	return hostname
+}
+
 func generateDSN(args argumentList) string {
 	// Format query parameters
 	query := url.Values{}
@@ -66,6 +78,8 @@ func generateDSN(args argumentList) string {
 		log.Info("Socket parameter is defined, ignoring host and port parameters")
 		return fmt.Sprintf("%s:%s@unix(%s)/%s?%s", args.Username, args.Password, args.Socket, args.Database, query.Encode())
 	}
+
+	args.Hostname = hostnameIpv6Formater(args.Hostname)
 
 	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?%s", args.Username, args.Password, args.Hostname, args.Port, args.Database, query.Encode())
 }

--- a/src/mysql.go
+++ b/src/mysql.go
@@ -43,17 +43,6 @@ type argumentList struct {
 	ShowVersion            bool   `default:"false" help:"Print build information and exit"`
 }
 
-func hostnameIpv6Formater(hostname string) (string) {
-	// Format backend parameters
-	ip := net.ParseIP(hostname)
-
-	if ip.To16() != nil {
-		return fmt.Sprintf("[%v]", ip)
-	}
-
-	return hostname
-}
-
 func generateDSN(args argumentList) string {
 	// Format query parameters
 	query := url.Values{}
@@ -79,9 +68,10 @@ func generateDSN(args argumentList) string {
 		return fmt.Sprintf("%s:%s@unix(%s)/%s?%s", args.Username, args.Password, args.Socket, args.Database, query.Encode())
 	}
 
-	args.Hostname = hostnameIpv6Formater(args.Hostname)
+	// Convert hostname and port to DSN address format
+	mysqlURL := net.JoinHostPort(args.Hostname, strconv.Itoa(args.Port))
 
-	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?%s", args.Username, args.Password, args.Hostname, args.Port, args.Database, query.Encode())
+	return fmt.Sprintf("%s:%s@tcp(%s)/%s?%s", args.Username, args.Password, mysqlURL, args.Database, query.Encode())
 }
 
 var (

--- a/src/mysql.go
+++ b/src/mysql.go
@@ -3,12 +3,12 @@ package main
 
 import (
 	"fmt"
+	"net"
 	"net/url"
 	"os"
 	"runtime"
 	"strconv"
 	"strings"
-	"net"
 
 	"github.com/newrelic/infra-integrations-sdk/data/attribute"
 

